apps/passport-client/src/user.ts

```diff
@@ -80,12 +80,13 @@ export function findUserIdentityV4PCD(
 
 export function findIdentityV4PCD(
   pcds: PCDCollection,
-  v4Commitment: string | undefined | null
+  v4Commitment?: string | undefined | null
 ): SemaphoreIdentityV4PCD | undefined {
   for (const pcd of pcds.getPCDsByType(SemaphoreIdentityV4PCDTypeName)) {
     if (
       (pcd as SemaphoreIdentityV4PCD).claim.identity.commitment.toString() ===
-      v4Commitment
+        v4Commitment ||
+      v4Commitment === undefined
```

Comment:
Why is this needed for V4 and not V3?

Reply:
Oh, I think this was because of my suggestion to centralize here.  This isn't actually the place where the "pick the first" fallback is encoded.  I don't mind if it becomes that, though I suggest explicitly documenting what a missing v4Commitment means, and why.  I.e. it means pick the first, because we use that to identify the user's primary identity.